tests/RobotFramework/tests/tedge_connect/tedge_connet_mqtt_service.robot

```diff
@@ -0,0 +1,55 @@
+*** Settings ***
+Resource            ../../resources/common.resource
+Library             DateTime
+Library             Cumulocity
+Library             ThinEdgeIO
+
+Test Teardown       Get Logs
+
+
+*** Test Cases ***
+Connect to Cumulocity MQTT Service endpoint
+    ${DEVICE_SN}=    Setup    connect=${False}
+    Execute Command    tedge config set c8y.tenant_id t37070943
+    Execute Command    tedge config set c8y.mqtt_service.enabled true
+    Execute Command    tedge config set c8y.mqtt_service.topics 'sub/topic,demo/topic'
+    Execute Command    tedge connect c8y
+
+    # TODO: Subscribing to test/topic from another client
+    Execute Command    tedge mqtt pub c8y-mqtt/test/topic '"hello"'
+    # TODO: Validate message received on test/topic on the other client
+
+    Sleep    1s
+
+Connect to Cumulocity MQTT Service endpoint basic auth
+    ${DEVICE_SN}=    Setup    register=${False}
+
+    Execute Command    tedge config set device.id ${DEVICE_SN}
+    Execute Command    tedge config set c8y.url "${C8Y_CONFIG.host}"
+    Execute Command    tedge config set c8y.mqtt_service.enabled true
+
+    Execute Command
+    ...    cmd=printf '[c8y]\nusername = "%s"\npassword = "%s"\n' '${C8Y_CONFIG.username}' '${C8Y_CONFIG.password}' > /etc/tedge/credentials.toml
+    Execute Command    tedge config set c8y.auth_method basic
+
+    Execute Command    tedge connect c8y
+
+    # TODO: Subscribing to test/topic from another client
+    Execute Command    tedge mqtt pub c8y-mqtt/test/topic '"hello"'
+    # TODO: Validate message received on test/topic on the other client
+
+    Sleep    1s
+
+Connect to Cumulocity MQTT Service endpoint builtin bridge
+    ${DEVICE_SN}=    Setup    connect=${False}
+    Execute Command    tedge config set mqtt.bridge.built_in true
+    Execute Command    tedge config set c8y.tenant_id t37070943
+    Execute Command    tedge config set c8y.mqtt_service.enabled true
+    Execute Command    tedge config set c8y.mqtt_service.topics 'sub/topic,demo/topic'
```

Comment:
There's something really weird with multiple subscriptions on MQTT service. MQTT service is rejecting subscription requests with multiple topics from the built-in bridge, which is why this test case was failing. Here is the same reproduced with `mosquitto_sub` client:

```sh
$ mosquitto_sub -v -h albin.latest.stage.c8y.io -p 9883 -i albin-123 --key device.key --cert device.crt --cafile c8y-root.crt -u t37070943 -t sub/topic -t demo/topic
All subscription requests were denied.
```

Once you limit the subscription to a single topic, it works. This test also works when there is only one subscription topic.

But to my surprise, the mosquitto bridge with multiple subscriptions works just fine (as done in the first test case).

Reply:
This was discussed offline, but the outcome was that it seems to be a current limitation of the MQTT Service, however we provided the appropriate RnD team that supporting multiple subscriptions would mean that the MQTT service would be more compatible with different MQTT tooling (and what users expect when interacting with such as service)

Reply:
It is already working with mosquitto bridge as it sends separate subscription requests for each bridge rule. The same behaviour was replicated in built-in bridge as well with [584b438](https://github.com/thin-edge/thin-edge.io/pull/3707/commits/584b43856836d724d072fa0b268568131f0313a1)